File: bst/compareTrees.ts
import { BSTNode } from "./bst";

function compareTrees(node1: BSTNode<number>|undefined, node2: BSTNode<number>|undefined):boolean{
 if (node1 === undefined && node2 === undefined){
  return true
 }
 if (node1 === undefined || node2 === undefined){
  return false
 } 
 if(node1.value !== node2.value) {
  return false
 }
 return compareTrees(node1.left, node2.left) && compareTrees(node1.right, node2.right) 
}

File: bst/compare-trees/compare_trees.py
def compare_trees(node1, node2):
  if node1 == None == node2:
    return True
  if node1 == None or node2 == None:
    return False
  if node1.value != node2.value:
    return False
  return compare_trees(node1.left, node2.left) and compare_trees(node1.right, node2.right)